big_o/test/test_big_o.py

```diff
@@ -78,3 +78,66 @@ def dummy_quadratic_function(n):
                 n_timings=10,
             )
             self.assertEqual(class_, res_class.__class__)
+
+    def test_big_o_return_raw_data_default(self):
+        def dummy_linear_function(n):
+            for _ in range(n):
+                # Dummy operation with constant complexity.
+                8282828 * 2322
+
+        _, fitted = big_o.big_o(
+            dummy_linear_function,
+            datagen.n_,
+            min_n=100,
+            max_n=10000,
+            n_measures=25,
+            n_repeats=10,
+            n_timings=10)
+
+        for _, v in fitted.items():
+            self.assertIsInstance(v, np.float64)
+
+    def test_big_o_return_raw_data_false(self):
+        def dummy_linear_function(n):
+            for _ in range(n):
+                # Dummy operation with constant complexity.
+                8282828 * 2322
+
+        _, fitted = big_o.big_o(
+            dummy_linear_function,
+            datagen.n_,
+            min_n=100,
+            max_n=10000,
+            n_measures=25,
+            n_repeats=10,
+            n_timings=10,
+            return_raw_data=False)
+
+        for _, v in fitted.items():
+            self.assertIsInstance(v, np.float64)
+
+    def test_big_o_return_raw_data_true(self):
+        def dummy_linear_function(n):
+            for _ in range(n):
+                # Dummy operation with constant complexity.
+                8282828 * 2322
```

Comment:
Can you please turn this into `time.sleep(0.05)` or something like that, so that you can check that each measure is `>= 0.05 * n_repeats * n_timings`? The test is weak because it does not check the actual content of `measures` and `times`

Reply:
I improved the tests to check the contents of `measures` and `times`. I didn't include `n_timings` to the assertion of `times` since it is not cumulative.